clippy.toml

```
@@ -1 +1,7 @@
 stack-size-threshold = 102400
+
+disallowed-types = [
+    { path = "std::sync::Mutex", reason = "parking_lot offers simpler, faster, and poison-free synchronization primitives", replacement = "parking_lot::Mutex" },
+    { path = "std::sync::RwLock", reason = "parking_lot offers simpler, faster, and poison-free synchronization primitives", replacement = "parking_lot::RwLock" },
+    { path = "std::sync::Condvar", reason = "parking_lot offers simpler, faster, and poison-free synchronization primitives", replacement = "parking_lot::Condvar" },
```

```suggestion
    { path = "std::sync::Mutex", reason = "parking_lot offers simpler synchronization primitives", replacement = "parking_lot::Mutex" },
    { path = "std::sync::RwLock", reason = "parking_lot offers simpler synchronization primitives", replacement = "parking_lot::RwLock" },
    { path = "std::sync::Condvar", reason = "parking_lot offers simpler synchronization primitives", replacement = "parking_lot::Condvar" },
```

We can't say for sure if they're faster without putting serious work to quantify this and we can't rely on folklore as it may very well be outdated. Simpler is somewhat synonymous with poison-free too.

---

Sure, appreciate your veracious attitude.